pkg/controller/servicebindingrequest/binder.go

```diff
@@ -44,9 +44,19 @@ func (b *Binder) search() (*unstructured.UnstructuredList, error) {
 		Version:  b.sbr.Spec.ApplicationSelector.Version,
 		Resource: b.sbr.Spec.ApplicationSelector.Resource,
 	}
-	matchLabels := b.sbr.Spec.ApplicationSelector.MatchLabels
-	opts := metav1.ListOptions{
-		LabelSelector: labels.Set(matchLabels).String(),
+
+	opts := metav1.ListOptions{}
+
+	// If Application name is present
+	if b.sbr.Spec.ApplicationSelector.ResourceRef != nil {
+		opts = metav1.ListOptions{
+			FieldSelector: b.sbr.Spec.ApplicationSelector.ResourceRef,
+		}
+	} else {
```

Comment:
So, at this point, both labels and resourceRef is optional. Please ensure we handle the scenario where nothing is provided - it should be a "requeue on error" - ish thing.

Reply:
right.

Reply:
What would the desired behaviour be in the case `ResourceRef` doesn't match `matchLabels`?

Reply:
@isutton In that case, the `ResourceRef` takes precedence. For now, the application matching the `matchLabels` criteria is ignored.